### PR TITLE
Enrich `evar_map` printer with future goals stack

### PR DESCRIPTION
**Pull request by maximedenes**
This is useful for debugging.

### CODE REVIEW

**Comment by maximedenes** [comment]
@ppedrot do you mind taking this one?

File: engine/evd.mli [hunk 1]
```diff
@@ -371,12 +371,16 @@ module FutureGoals : sig
...
 
+  type stack
+
   val map_filter : (Evar.t -> Evar.t option) -> t -> t
   (** Applies a function on the future goals *)
 
   val filter : (Evar.t -> bool) -> t -> t
   (** Applies a filter on the future goals *)
 
+  val pr_stack : stack -> Pp.t
```
Comment (ppedrot) [inline engine/evd.mli:382]: Can't you just export a function taking an evarmap instead of having to export the `stack` type?

Comment (maximedenes) [reply engine/evd.mli]: Done.

**Review by ppedrot — APPROVED** [review]
LGTM.

**Comment by ppedrot** [comment]
Will merge soon, it's a debug feature so it shouldn't need a lot of thoughts.

**Pull request merged**